### ISSUE TITLE
UpdateMatchedAndConditionNames in BulkOperation are ignored for PostgreSql

### ISSUE THREAD

**Issue by nothings-more** [zzzprojects/Dapper-Plus#103]
Hello,

I'm trying to extend bulk update operation using `UpdateMatchedAndConditionNames` , but looks like this setting is not affect resulting SQL generated for PostgreSql provider, while it works for SqlServer provider as expected.

For example, consider having an entity with 2 properties: `Id` (key) and `Version`. Following code
```csharp
connection.UseBulkOptions(bulkOperation =>
            {
                bulkOperation .UpdateMatchedAndConditionNames = new List<string> { "Version" };
            }).BulkUpdate(new[] { entity });
```
produces `MERGE` statement for SqlServer provider, with expected additional condition for "Version", i.e. in SqlServer bulk update will be performed only for those rows in DB where "Id" and "Version" are equal to corresponding properties of entity to update.

For PostgreSql provider, though, the same C# code produces `UPDATE` with only "Id" comparison included in `WHERE` clause, i.e. comparison of "Version" is ignored.

I've sent a sample project with ability to test both providers to info@zzzprojects.com

Also, looks like there is a similar issue for `MergeMatchedAndConditionNames` property for bulk operations

Please let me know if you need any additional info on the issue.

Thanks in advance,
Alex

**Comment by JonathanMagnan** [comment]
Helloo @nothings-more ,

The v4.0.21 has been released.

Could you try it and confirm us that everything is now working for the option `UpdateMatchedAndConditionNames`?

Best Regards,

Jon

**Comment by nothings-more** [comment]
@JonathanMagnan , yes, looks like the issue is fixed in 4.0.21
Thank you
